lib/metadataTypes/MobileMessage.js

```diff
@@ -0,0 +1,550 @@
+'use strict';
+
+const TYPE = require('../../types/mcdev.d');
+const MetadataType = require('./MetadataType');
+const Util = require('../util/util');
+const File = require('../util/file');
+const cache = require('../util/cache');
+
+/**
+ * MobileMessage MetadataType
+ *
+ * @augments MetadataType
+ */
+class MobileMessage extends MetadataType {
+    /**
+     * Retrieves Metadata of Mobile Keywords
+     *
+     * @param {string} retrieveDir Directory where retrieved metadata directory will be saved
+     * @param {void} [_] unused parameter
+     * @param {void} [__] unused parameter
+     * @param {string} [key] customer key of single item to retrieve
+     * @returns {Promise.<TYPE.MetadataTypeMapObj> | void} Promise of metadata
+     */
+    static retrieve(retrieveDir, _, __, key) {
+        this.upgradeGetBulk();
+        if (key && key.startsWith('id:')) {
+            // if key starts with id: remove it to be compatible with other legacy API types (MetadataType.postCreateTasks_legacyApi)
```

Comment:
it randomly starts with id sometimes?

Reply:
nope, the generic script we use for the legacy API needs to be able to pass that in - because legacy API needs IDs and we have that there already and can then cache the right thing